server/src/main/java/org/opensearch/http/AbstractHttpServerTransport.java

```diff
@@ -352,17 +370,55 @@ protected void serverAcceptedChannel(HttpChannel httpChannel) {
      * @param httpChannel that received the http request
      */
     public void incomingRequest(final HttpRequest httpRequest, final HttpChannel httpChannel) {
-        handleIncomingRequest(httpRequest, httpChannel, httpRequest.getInboundException());
+        try (ThreadContext.StoredContext ignore = threadPool.getThreadContext().stashContext()) {
+            // TODO: Add support for parsing the otel incoming tracer.
+            final SpanScope httpRequestSpanScope = telemetryTracer.startSpan(
+                SPAN_NAME_INCOMING_HTTP_SERVER_REQUEST,
+                buildSpanAttributes(httpRequest)
+            );
+            httpRequestSpanScope.addSpanAttribute(SPAN_ATTR_KEY_REQ_INBOUND_EX, httpRequest.getInboundException() != null);
+            HttpChannel traceableHttpChannel = new TraceableHttpChannel(httpChannel, httpRequestSpanScope);
+            handleIncomingRequest(httpRequest, traceableHttpChannel, httpRequest.getInboundException());
+        }
+    }
+
+    private Attributes buildSpanAttributes(HttpRequest httpRequest) {
+        Attributes attributes = Attributes.create()
+            .addAttribute(SPAN_ATTR_KEY_URI, httpRequest.uri())
+            .addAttribute(SPAN_ATTR_KEY_METHOD, httpRequest.method().name())
+            .addAttribute(SPAN_ATTR_KEY_VERSION, httpRequest.protocolVersion().name());
+        populateHeader(httpRequest, attributes);
+        return attributes;
+    }
+
+    private void populateHeader(HttpRequest httpRequest, Attributes attributes) {
+        HEADERS_TO_BE_POPULATED.forEach(x -> {
+            if (httpRequest.getHeaders() != null && httpRequest.getHeaders().get(x) != null) {
+                attributes.addAttribute(x, Strings.collectionToCommaDelimitedString(httpRequest.getHeaders().get(x)));
+            }
+        });
     }
 
     // Visible for testing
     void dispatchRequest(final RestRequest restRequest, final RestChannel channel, final Throwable badRequestCause) {
+        final SpanScope spanScope = telemetryTracer.startSpan(SPAN_NAME_REST_REQUEST);
+        spanScope.addSpanAttribute(SPAN_ATTR_KEY_BAD_REQ_PARAM, badRequestCause != null);
```

Comment:
The `badRequestCause` would end up as `scope.error`, we don't need to duplicate it here

Reply:
We are trying the request without Parameters in this case don't know how it would end up.

Reply:
> Request is still be tried without Parameters in this case don't know how it would end up.

It could fail or succeed but that will be reflected on span termination (success or error)